### DISCUSSION TITLE
Updating cart using AlpineJS only

### DISCUSSION THREAD

**Discussion by rrrrando** [duncanmcclean/simple-commerce#1036] [Help] [answered]
Hey!

I am trying to update item quantity in the cart view using AJAX via AlpineJS, but some of the needed data like `total_including_tax` and variant images are not in the api response. I think i should create a custom endpoint for that, but I'm stuck on what would be the best way to get all the cart data. If someone could point me to the right direction, it would be much appreciated.

**Comment by duncanmcclean** [comment, marked as answer]
While Simple Commerce _does_ have a dedicated endpoint to return the cart & its line items (GET `!/simple-commerce/cart`), it won't include the `total_including_tax` bit you're after (since it's a tag, not actually part of the augmentation).

If you need that (or any other bits of information not returned by that endpoint), you'd have to make your own controller & route:

```php
// app/Http/Controllers/CartController.php
<?php

namespace App\Http\Controllers;

use DuncanMcClean\SimpleCommerce\Orders\Cart\Drivers\CartDriver;
use Illuminate\Http\Request;

class CartController extends Controller
{
    use CartDriver;

    public function __invoke(Request $request)
    {
        $order = $this->getCart();

        if (! $order) {
            abort(404);
        }

        return response()->json([
            'order_number' => $order->orderNumber(),
            'items' => $order->lineItems()
                ->map(function ($lineItem) {
                    return array_merge($lineItem->toArray(), [
                        'total_including_tax' => $lineItem->totalIncludingTax(),
                    ]);
                })
                ->all(),
            // ... anything else you want to return
            // see the docs for reference: https://simple-commerce.duncanmcclean.com/php-api#content-orders
        ]);
    }
}
```

```php
// routes/web.php

use App\Http\Controllers\CartController;

Route::get('/cart-json', CartController::class);
```

**Reply by rrrrando** [reply]
Thanks! Thats exactly what i was looking for!
